### DISCUSSION TITLE
I need to shift the text on the left half 55 pixels to the right, and the text on the right half 35 pixels to the left in a double-page scan

### DISCUSSION THREAD

**Discussion by oavaldezi** [ImageMagick/ImageMagick#2016] [General]
In several double-page scans, I need to shift the text on the left half 55 pixels to the right, and the text on the right half 35 pixels to the left. Thanks for your tips!
![imagen008](https://user-images.githubusercontent.com/2160804/81875460-88452d00-953d-11ea-89bb-a6d7c1e4b6ef.jpg)
![imagen009](https://user-images.githubusercontent.com/2160804/81875464-8da27780-953d-11ea-8a37-d43b9e7f2f3d.jpg)

**Comment by fmw42** [comment]
Crop the page into two parts horizontally. Then do -trim to remove all the white borders. Then pad them white the amount of white you want. Then append them together horizontally with any white padding you want and any spacer between them. See +append or +smush along with -border or -extent. Always provide your ImageMagick version and platform, since syntax may differ.

**Reply by oavaldezi** [reply]
Thanks for your tip. I learned about `-trim`, but unfortunately, it doesn't quite do what I need, because not all pages have the identical borders, either horizontally or vertically.
I'm giving -copy _geometry offset_ a try, but instead of offsetting the selected geometry, I end up with a white half page. For example, I want `-copy 1523x2252+0+0 +55+0` to shift the left half 55 pixels to the right, but I end up with a white left half!
![imagen-8b](https://user-images.githubusercontent.com/2160804/81885054-bbe08100-9556-11ea-8c4f-c747f8adbd56.jpg)

**Reply by fmw42** [reply]
You never answered about your IM version and platform. Assuming IM 6 and Unix, you can do the following to make the center spacing 300 pixels and the outside margin 100 pixels with the text aligned to the top (north).

```
center=300
pad=100
convert page.jpg -crop 50x100% +repage \
-fuzz 15% -trim +repage \
-gravity north -background white +smush $center \
-bordercolor white -border $pad \
page_new.jpg

```


![page_new](https://user-images.githubusercontent.com/13458218/81886478-a452c980-9551-11ea-9eb7-1e1057709603.jpg)


Adjust the center and pad as desired. Or you know how big you want the page to be, you can do (making the center spacing 300 and output size 3000x2250):

```
dims=3000x2250
center=300
convert page.jpg -crop 50x100% +repage \
-fuzz 15% -trim +repage \
-gravity north -background white +smush $center \
-gravity north -background white -extent $dims \
page_new2.jpg

```

![page_new2](https://user-images.githubusercontent.com/13458218/81886654-17f4d680-9552-11ea-8424-99409154a160.jpg)

**Reply by oavaldezi** [reply]
Sorry. ImageMagick 7.0.10-10 on Windows 10. I'll play around with your suggestions. Thanks!

**Reply by fmw42** [reply]
Change convert to magick and replace the end of line \  with ^ for Windows and IM 7. If in a .bat file replace % with %%. Also variables are defined and used differently in Windows and Unix. See https://imagemagick.org/Usage/windows/

**Reply by oavaldezi** [reply]
Didn't work, because of differing margins around pages. I've attached a before and after.

Why do I end up with a white left half when I `-copy 1523x2252+0+0 +55+0` , instead of shifting it 55 pixels to the right?

![imagen007](https://user-images.githubusercontent.com/2160804/81888720-2990ab00-955f-11ea-89f3-cf2cd81fc889.jpg)
![imagen007-a](https://user-images.githubusercontent.com/2160804/81888725-2d243200-955f-11ea-8155-f9ee1558e771.jpg)

**Reply by fmw42** [reply]
I have never used -copy. But if you move a section, you have to fill where it was with something. That is usually comes from -background.

What is wrong with my code? Why do you need to preserve the margins? Can you not just set the desired page width or pad as in my first example by some given amount. If you have to preserve the margins, then it can be done with extra processing to determine how big the margins are.

**Reply by oavaldezi** [reply]
Thank you for your interest and help. I'm trying not only to preserve but also to restore the original pages' layout.

In the end, I couldn't get `-copy` to work correctly: it always copied white pixels, so I ended up using `-splice` to widen the leftmost and rightmost margins, and `-chop` to decrease the width of the center gutter. Here´s my complete command, and I've zipped a sample starting image and the results:

`magick imagen-0008.pbm -resize 3046x2252 -gravity center -extent 3046x2252 -gravity west -splice 55x0 -gravity east -splice 35x0 -fill White -draw "rectangle 1478,0,1678,2252" -gravity center -chop 90x0 -crop 1523x2252,1523x2260+1613 -alpha off imagen-%03d.tiff`

`-crop 2x1@ +repage` didn't work correctly either, so I had to specify the regions to cut out. Again, thanks for your tips. They were a valuable part of my learning process.

[imagen-0008.zip](https://github.com/ImageMagick/ImageMagick/files/4630330/imagen-0008.zip)

**Reply by fmw42** [reply]
This does not make sense to me

> -crop 1523x2252,1523x2260+1613

crop only has 4 arguments WxH+X+Y

**Reply by oavaldezi** [reply]
According to the documentation:

_-crop geometry{@}{!}
Cut out one **or more** rectangular regions of the image._

So I figured if I wanted to crop more than one rectangular region, I needed to separate their geometries with commas, which should really be `-crop 1523x2252,1523x2252+1613`

**Reply by fmw42** [reply]
It does not work that way. You can only specify one set of WxH+X+Y. If you want multiple crops, it would be specified without +X+Y and you would get every block of WxH in the image. See https://imagemagick.org/Usage/crop/#crop_tile

**Reply by oavaldezi** [reply]
It did work for me! Try it with the sample imagen-0008.pbm file I zipped earlier. You should get two 1523x2252 files.

**Reply by fmw42** [reply]
Probably just saw it as -crop 1523x2250 and that only got your two images since the width was less than 2 x 1523.

Replace

`-crop 1523x2252,1523x2252+1613` with `-crop 1523x2252` and see if you get the same results as before

**Reply by oavaldezi** [reply]
Hmm... The second region I need to crop is offset 1613x0 pixels from the top left corner, which is why I specified the second geometry as 1523x2252+1613.

**Reply by fmw42** [reply]
I just do not think you can do it that way with two crops from one -crop with offsets.

**Reply by oavaldezi** [reply]
It worked for me. Give it a try!

**Reply by fmw42** [reply]
There is no difference in the results when using `-crop 1523x2252,1523x2260+1613` or `-crop 1523x2252`. ImageMagick is ignoring everything after the comma, so it only sees `-crop 1523x2252`

**Reply by oavaldezi** [reply]
> There is no difference in the results when using `-crop 1523x2252,1523x2260+1613` or `-crop 1523x2252`.

You are right about that, and it's documented: _If the x and y offsets are omitted, a set of tiles of the specified geometry, covering the entire input image, is generated. The rightmost tiles and the bottom tiles are smaller if the specified geometry extends beyond the dimensions of the input image._

Thank you for your time, and for helping me learn more about ImageMagick.
